Add --color option to kubectl cluster-info cmd

hypnoglow opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. If this is your first time, read our contributor guidelines https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
3. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
4. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What this PR does / why we need it**:

Currently, there is no way to disable colors in `kubectl cluster-info`
command output. This commit adds a new flag `--color` that solves
this, thus you can do `kubectl cluster-info --color=false`.

**Special notes for your reviewer**:

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
Add --color flag to kubectl cluster-info command
```

dixudx commented on pkg/kubectl/cmd/clusterinfo.go:
```diff
@@ -103,7 +112,7 @@ func (o *ClusterInfoOptions) Run() error {
 			return err
 		}
-		printService(o.Out, "Kubernetes master", o.Client.Host)
+		printService(o.Out, "Kubernetes master", o.Client.Host, !o.NoColor)
```
`!o.NoColor` is equal to `o.Color`

That's why I'd prefer to use `o.Color` directly.

dixudx commented on pkg/kubectl/cmd/clusterinfo.go:
```diff
@@ -61,15 +66,19 @@ func NewCmdClusterInfo(f cmdutil.Factory, ioStreams genericclioptions.IOStreams)
 		},
 	}
+
+	cmd.Flags().BoolVar(&o.NoColor, "no-color", false, "Disable colors in the output")
```
I'd prefer to use `--color` directly.

Please bind the default value in L65. 

And update as below,

```
cmd.Flags().BoolVar(&o.Color, "color", o.Color, "enable colors in the output")
```

hypnoglow replied:
Yeah, `--color` is much better, thanks!

dixudx reviewed (commented):
Nice add-on.

dixudx commented:
/ok-to-test
/lgtm

hypnoglow commented:
Changed formatting in L70-L74 in accordance with gofmt.

dixudx commented:
/lgtm

mengqiy commented on pkg/kubectl/cmd/clusterinfo.go:
```diff
@@ -58,10 +63,12 @@ type ClusterInfoOptions struct {
 	cmd := &cobra.Command{
-		Use:     "cluster-info",
+		Use: "cluster-info",
+		DisableFlagsInUseLine: true,
```
Why you need this?

hypnoglow replied:
I have no reason. I saw this widely used in other commands. I can remove this if we don't want this.

hypnoglow commented:
/retest

hypnoglow commented:
/retest

hypnoglow commented:
Rebased, as it was a bit stale

hypnoglow committed:
Add --no-color option to kubectl cluster-info cmd

Currently, there is no way to disable colors in `kubectl cluster-info`
command output. This commit adds a new flag `--no-color` that solves
this.

hypnoglow commented:
/remove-lifecycle rotten

hypnoglow commented:
/retest
